src/projectM-qt/qplaylistmodel.cpp

```diff
@@ -382,9 +382,12 @@ void QPlaylistModel::clear()
 
 void QPlaylistModel::clearItems()
 {
-	beginRemoveRows ( QModelIndex(), 0, rowCount()-1 );
-	m_projectM.clearPlaylist();
-	endRemoveRows();
+    if (rowCount() > 0)
+    {
+        beginRemoveRows(QModelIndex(), 0, rowCount() - 1);
+        m_projectM.clearPlaylist();
+        endRemoveRows();
```

Comment:
I dunno what's going on here but maybe we should always be calling `endRemoveRows`?

Reply:
As the model represents the playlist items, we need to call [`beginRemoveRows`](https://doc.qt.io/qt-6/qabstractitemmodel.html#beginRemoveRows) first to tell the model that we are going to change the underlying structure, then perform the operation (clearing the playlist) and call `endRemoveRows` to tell teh model that we're done removing the rows. You must only call `endRemoveRows` after a matching `beginRemoveRows` call. I've only added an `if` statement checking whether there are any rows because otherwise, the last row index would be -1, which is invalid and will cause an assertion to fail (first row must be larger or equal to last row). It's not critical, Qt release builds will simply ignore it, but it might as well crash the application.